sdk/typescript/src/domains/zeto.ts

```diff
@@ -156,4 +336,85 @@ export class ZetoInstance {
     });
     return this.paladin.pollForReceipt(txID, this.options.pollTimeout);
   }
+
+  async setERC20(from: string, data: ZetoSetERC20Params) {
+    const txID = await this.paladin.sendTransaction({
+      type: TransactionType.PUBLIC,
+      abi: zetoPublicAbi,
+      function: "setERC20",
+      to: this.address,
+      from,
+      data,
+    });
+    const result = await this.paladin.pollForReceipt(txID, POLL_TIMEOUT_MS);
+    if (result === undefined) {
+      throw new Error("Failed to set ERC20");
+    }
+    this.erc20 = data._erc20;
+    return result;
+  }
+
+  async deposit(from: string, data: ZetoDepositParams) {
+    // first approve the Zeto contract to draw the amount from our balance in the ERC20
+    const txID1 = await this.paladin.sendTransaction({
+      type: TransactionType.PUBLIC,
+      abi: erc20Abi,
+      function: "approve",
+      to: this.erc20,
+      from,
+      data: { value: data.amount, spender: this.address },
+    });
+    const result1 = await this.paladin.pollForReceipt(txID1, POLL_TIMEOUT_MS);
+    if (result1 === undefined) {
+      throw new Error("Failed to approve transfer");
+    }
+
+    const depositID = await this.paladin.prepareTransaction({
```

Comment:
Why do we manually prepare and then send, instead of just sending?

Reply:
at the moment Paladin has a domain-wide setting to either use a designated signer for all the private transactions, or use a hd wallet backed per-tx signing key. Here we need to use the `From` identity to actually sign this particular private transaction, while still allowing the rest of the `transfer` calls to be signed with a one-time key.

Reply:
talking with @awrichar further, there a `RequiredSigner` field in the response of the Prepare() domain function, which sounds like could be used to enforce the signer to be the `From` field identity.